hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/kafka/PropertiesResolver.java

```diff
@@ -62,49 +70,51 @@ static Properties resolveProperties(Map<String, String> options) {
         Properties properties = new Properties();
         properties.putAll(options);
 
-        properties(options, true).forEach(properties::putIfAbsent);
-        properties(options, false).forEach(properties::putIfAbsent);
+        withSerdeProperties(true, options, properties);
+        withSerdeProperties(false, options, properties);
 
         return properties;
     }
 
-    private static Map<String, String> properties(Map<String, String> options, boolean isKey) {
-        Map<String, String> properties = new HashMap<>();
-
+    private static void withSerdeProperties(
+            boolean isKey,
+            Map<String, String> options,
+            Properties properties
+    ) {
         String serializer = isKey ? KEY_SERIALIZER : VALUE_SERIALIZER;
         String deserializer = isKey ? KEY_DESERIALIZER : VALUE_DESERIALIZER;
 
         String format = options.get(isKey ? OPTION_KEY_FORMAT : OPTION_VALUE_FORMAT);
-        if (JAVA_FORMAT.equals(format)) {
+        if (format == null && isKey) {
+            properties.putIfAbsent(serializer, NULL_SERIALIZER);
+            properties.putIfAbsent(deserializer, NULL_DESERIALIZER);
```

Comment:
NullSerializer might be OK, we'll never try to write to the key. But NullDeserializer isn't, the key might be there, we just don't map it. I used `ByteArrayDeserializer` and `ByteArraySerializer`, these work with any data.